src/robot/running/arguments/typeconverter.py

```diff
@@ -39,12 +40,17 @@ def convert(self, positional, named):
     def _convert(self, name, value):
         if name not in self._argspec.types or not is_string(value):
             return value
+        if value.upper() == 'NONE':
+            return None
```

Comment:
Is moving these two lines related to enums? For me it feels a bit wrong to convert `NONE` string to `None` always if an argument has annotation. For example, if we have `def example(arg: str)`, we should pass all strings (and all other arguments) without changes. This conversions is ought to be safe if the annotation is a type we are going to convert and know wouldn't work with strings in the first place.